CONTRIBUTION.md

```diff
@@ -0,0 +1,39 @@
+Branching names:
+---
+Each branch must have a name in the following format with a proper prefix and number of Github's ticket it refers to:
+
+```
+prefix/000-branch-name
+``` 
+
+Allowed prefixes:
+* feature
+* bugfix
+* docs
+* test (when adding missing tests)
+* ...
+
+Commits:
+---
+Each commit (except for merge commits) must have a name in the following format:
+
+```
+000 - description of what was changed
```

Comment:
Including the ticket name in the commit description is quite unusual I must say.
There is extensive material on how to write a commit message, I'll find some reference

Reply:
https://chris.beams.io/posts/git-commit/ is one that seems inline with common practice

Reply:
Yes, the ticket name in the commit might be too painful, especially if we generally keep a one ticket to one PR association.

Reply:
I personally find it useful when I'm debugging code. It's easier to track a reason why that line was changed that way when you can link it to its task. It makes reading git history more comfortable as well. Otherways you need to find a task based on a merge commit name which sometimes (i.e. when you use feature branch) might be tricky to track.

Reply:
Is there a way to enforce this on GitHub using any sort of hooks?

Reply:
I'm not sure. There is for GIT for sure but it requires to install it manually. Anyway, I'm not sure if adding an automatic check would be useful for us.